cli/addressbook/address.go

```diff
@@ -0,0 +1,112 @@
+package addressbook
+
+import (
+	"fmt"
+
+	"github.com/AlecAivazis/survey/v2"
+	ethCommon "github.com/ethereum/go-ethereum/common"
+	"github.com/mitchellh/mapstructure"
+	"github.com/spf13/cobra"
+
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/helpers"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/types"
+)
+
+const (
+	// ethAddressCfgKey is an optional key passed in the config files.
+	ethAddressCfgKey = "eth_address"
+)
+
+// ImportSource is a source of imported address.
+type ImportSource struct {
+	Data string
+}
+
+type addressConfig struct {
+	EthAddress string `mapstructure:"eth_address,omitempty"`
+}
+
+type Address struct {
+	address types.Address
+	cfg     *addressConfig
+}
+
+func newAddress(address types.Address, cfg *addressConfig) (*Address, error) {
+	return &Address{address: address, cfg: cfg}, nil
+}
+
+type AddressFactory struct{}
+
+func (a *AddressFactory) unmarshalConfig(raw map[string]interface{}) (*addressConfig, error) {
+	if raw == nil {
+		// No eth_address defined.
+		return &addressConfig{}, nil
+	}
+
+	var cfg addressConfig
+	if err := mapstructure.Decode(raw, &cfg); err != nil {
+		return nil, err
+	}
+	return &cfg, nil
+}
+
+func (a *AddressFactory) Load(_ string, rawCfg map[string]interface{}, rawAddr string) (*Address, error) {
+	cfg, err := a.unmarshalConfig(rawCfg)
+	if err != nil {
+		return nil, err
+	}
+
+	addr, _, err := helpers.ResolveEthOrOasisAddress(rawAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	return newAddress(*addr, cfg)
+}
+
+func (a *AddressFactory) Add(cfg map[string]interface{}, src *ImportSource) (*Address, error) {
```

Comment:
Will you need anything more than a simple string for `src` (it could also be named `address`)?